api/core/model_runtime/model_providers/ollama/llm/llm.py

```diff
@@ -544,56 +630,77 @@ def get_customizable_model_schema(self, model: str, credentials: dict) -> AIMode
                     type=ParameterType.INT,
-                    help=I18nObject(en_US="Sets how far back for the model to look back to prevent repetition. "
-                                          "(Default: 64, 0 = disabled, -1 = num_ctx)"),
-                    min=-1
+                    help=I18nObject(
+                        en_US="Sets how far back for the model to look back to prevent repetition. "
+                        "(Default: 64, 0 = disabled, -1 = num_ctx)"
+                    ),
+                    min=-1,
                 ),
                 ParameterRule(
-                    name='tfs_z',
+                    name="tfs_z",
                     label=I18nObject(en_US="TFS Z"),
                     type=ParameterType.FLOAT,
-                    help=I18nObject(en_US="Tail free sampling is used to reduce the impact of less probable tokens "
-                                          "from the output. A higher value (e.g., 2.0) will reduce the impact more, "
-                                          "while a value of 1.0 disables this setting. (default: 1)"),
-                    precision=1
+                    help=I18nObject(
+                        en_US="Tail free sampling is used to reduce the impact of less probable tokens "
+                        "from the output. A higher value (e.g., 2.0) will reduce the impact more, "
+                        "while a value of 1.0 disables this setting. (default: 1)"
+                    ),
+                    precision=1,
                 ),
                 ParameterRule(
-                    name='seed',
+                    name="seed",
                     label=I18nObject(en_US="Seed"),
                     type=ParameterType.INT,
-                    help=I18nObject(en_US="Sets the random number seed to use for generation. Setting this to "
-                                          "a specific number will make the model generate the same text for "
-                                          "the same prompt. (Default: 0)"),
+                    help=I18nObject(
+                        en_US="Sets the random number seed to use for generation. Setting this to "
+                        "a specific number will make the model generate the same text for "
+                        "the same prompt. (Default: 0)"
+                    ),
                 ),
                 ParameterRule(
-                    name='format',
+                    name="keep_alive",
+                    label=I18nObject(en_US="Keep Alive"),
+                    type=ParameterType.STRING,
+                    help=I18nObject(
+                        en_US="Sets how long the model is kept in memory after generating a response. "
+                        "This can be a duration string (e.g., '10m' or '24h'), a number in seconds "
+                        "(e.g., 3600), a negative number to keep the model loaded indefinitely, or '0' "
```

Comment:
Hi, I found that number without units could not be accepted, there is an error raised
<img width="760" alt="image" src="https://github.com/langgenius/dify/assets/45712896/618ec789-1833-484b-8d9c-0f56fdba5bbb">
And I've also checked the source code of Ollama, it uses json.Unmarshal to parse requests and `keep_alive` is defined as a time.Duration object, it requires at least one unit to be unmarshalled
<img width="846" alt="image" src="https://github.com/langgenius/dify/assets/45712896/c3f8cc7d-33d1-4a5c-9d9d-d56113960cda">
Maybe we could optimize this description or just append `s` to `keep_alive` if there is not a unit?

Reply:
I have updated the helper text.